server/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -462,11 +464,12 @@ public final void onShardFailure(final int shardIndex, @Nullable SearchShardTarg
      * @param result the result returned form the shard
      * @param shardIt the shard iterator
      */
-    private void onShardResult(Result result, SearchShardIterator shardIt) {
+    protected void onShardResult(Result result, SearchShardIterator shardIt) {
```

Comment:
nit: I think both `results` and `onShardResult()` can stay as private.

Reply:
Thanks, I pushed https://github.com/elastic/elasticsearch/pull/51708/commits/662972cf2de9b2ad36077b886c4930c8d2a08468